File: codigo/backend/go-backend/models/usuario.go
package models

type Usuario struct {
    ID        uint   `gorm:"primaryKey"`
    Endereco  string `gorm:"type:varchar(255);not null"`
    TipoUser  string `gorm:"type:varchar(50);not null"`
    Email     string `gorm:"type:varchar(255);not null"`
    Senha     string `gorm:"type:varchar(255);not null"`
}


func (Usuario) TableName() string {
    return "usuarios" // Nome da tabela no banco de dados
}

func (u *Usuario) ToMap() map[string]interface{} {
    return map[string]interface{}{
        "id":       u.ID,
        "endereco": u.Endereco,
        "tipouser": u.TipoUser,
        "email":    u.Email,
        "senha":    u.Senha,
    }
}

func (u *Usuario) String() string {
    return "<Usuario " + u.Email + ">"
}

File: codigo/backend/sap-ecc-emulator/src/models/usuario.py
from db import db

class Usuario(db.Model):
    __tablename__ = 'usuarios'

    id = db.Column(db.Integer, primary_key=True)
    endereco = db.Column(db.String(255), nullable=False)
    tipouser = db.Column(db.String(50), nullable=False)
    email = db.Column(db.String(255), nullable=False)
    senha = db.Column(db.String(255), nullable=False)

    def to_dict(self):
        return {
            'id': self.id,
            'endereco': self.endereco,
            'tipouser': self.tipouser,
            'email': self.email,
            'senha': self.senha
        }

    def __repr__(self):
        return f'<Usuario {self.email}>'
